client/homebrew/utils/versionHistory.js

```diff
@@ -0,0 +1,123 @@
+export const HISTORY_PREFIX = 'HOMEBREWERY-HISTORY';
+export const HISTORY_SLOTS = 5;
+
+const DEFAULT_HISTORY_SAVE_DELAYS = {
+ 	'0' : 0,           // 0 minutes (if not specified)
+	'1' : 2,			// 2 minutes
+	'2' : 10,			// 10 minutes
+	'3' : 60,			// 60 minutes
+	'4' : 12 * 60,     // 12 hours
+	'5' : 2 * 24 * 60  // 2 days
+};
+
+const DEFAULT_GARBAGE_COLLECT_DELAY = 28 * 24 * 60; // 28 days
+
+const DEFAULT_STORED_BREW = {
+	shareId  : 'default_brew',
+	expireAt : '2000-01-01T00:00:00.000Z'
+};
+
+const HISTORY_SAVE_DELAYS = global.config?.historyData?.HISTORY_SAVE_DELAYS ?? DEFAULT_HISTORY_SAVE_DELAYS;
+const GARBAGE_COLLECT_DELAY = global.config?.historyData?.GARBAGE_COLLECT_DELAY ?? DEFAULT_GARBAGE_COLLECT_DELAY;
+
+
+
+function getKeyBySlot(brew, slot){
+	return `${HISTORY_PREFIX}-${brew.shareId}-${slot}`;
+};
+
+function getVersionBySlot(brew, slot){
+	// Read stored brew data
+	// - If it exists, parse data to object
+	// - If it doesn't exist, pass default object
+	const key = getKeyBySlot(brew, slot);
+	const storedVersion = localStorage.getItem(key);
+	const output = storedVersion ? JSON.parse(storedVersion) : { ...DEFAULT_STORED_BREW, ...brew };
+	return output;
+};
+
+function updateStoredBrew(brew, slot=0){
+	const archiveBrew = {};
+
+	// Data from brew to be stored
+	archiveBrew.title = brew.title;
+	archiveBrew.text = brew.text;
+	archiveBrew.style = brew.style;
+	archiveBrew.version = brew.version;
+	archiveBrew.shareId = brew.shareId;
+
+	// Calculated values
+	archiveBrew.savedAt = brew?.savedAt || new Date();
+	archiveBrew.expireAt = new Date();
+	archiveBrew.expireAt.setMinutes(archiveBrew.expireAt.getMinutes() + HISTORY_SAVE_DELAYS[slot]);
+
+	// Store data
+	const key = getKeyBySlot(brew, slot);
+	localStorage.setItem(key, JSON.stringify(archiveBrew));
+	return;
+};
```

Comment:
Not critical, but would you be opposed to just setting the object directly?

```suggestion
function updateStoredBrew(brew, slot = 0) {
  const archiveBrew = {
    title    : brew.title,
    text     : brew.text,
    style    : brew.style,
    version  : brew.version,
    shareId  : brew.shareId,
    savedAt  : brew?.savedAt || new Date(),
    expireAt : new Date()
  };

  archiveBrew.expireAt.setMinutes(archiveBrew.expireAt.getMinutes() + HISTORY_SAVE_DELAYS[slot]);

  const key = getKeyBySlot(brew, slot);
  localStorage.setItem(key, JSON.stringify(archiveBrew));
}

```